machines-endpoints/pkg/machines-endpoints/main.go

```diff
@@ -346,6 +350,72 @@ func localHTTPClient() *http.Client {
 	}
 }
 
+func createMachinesList(machines []Machine) (string, error) {
+	type Machine struct {
+		Serial string `json:"serial"`
+		BmcIP  string `json:"bmc_ip"`
+		NodeIP string `json:"ipv4"`
+		Role   string `json:"role"`
+		State  string `json:"state"`
+	}
+	var ml []Machine
+
+	for _, machine := range machines {
+		var m Machine
+		if machine.Spec.BMC.IPv4 == "" {
+			continue
+		}
+		m.Serial = machine.Spec.Serial
+		m.BmcIP = machine.Spec.BMC.IPv4
+		m.NodeIP = machine.Spec.IPv4[0]
+		m.Role = machine.Spec.Role
+		m.State = machine.Status.State
+		ml = append(ml, m)
+	}
+
+	byteJSON, err := json.Marshal(ml)
+	if err != nil {
+		return "", err
+	}
+	return string(byteJSON), nil
+}
+
+var prevMachinesList string
+
+func (c client) updateBmcLogCollectorConfigMap(ctx context.Context, machinesList string) error {
+	if prevMachinesList == machinesList {
+		return nil
+	}
+	prevMachinesList = machinesList
+
+	addresses := make(map[string]string)
+	addresses["machineslist.json"] = machinesList
+
+	ns, _, err := c.kubeConfig.Namespace()
+	if err != nil {
+		return err
+	}
+
+	configMap := corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: bmcLogCollectorConfigMapName,
+		},
+		Data: addresses,
+	}
```

Comment:
Variable name `addresses` is not appropriate.

```golang
	data := make(map[string]string)
	data["machineslist.json"] = machinesList

	ns, _, err := c.kubeConfig.Namespace()
	if err != nil {
		return err
	}

	configMap := corev1.ConfigMap{
		ObjectMeta: metav1.ObjectMeta{
			Name: bmcLogCollectorConfigMapName,
		},
		Data: data,
	}
```
or
```golang
	configMap := corev1.ConfigMap{
		ObjectMeta: metav1.ObjectMeta{
			Name: bmcLogCollectorConfigMapName,
		},
		Data: map[string]string{
			"machineslist.json": machinesList,
		},
	}
```

Reply:
Changed